src/smart_building_rating_calculator/flex_archetype.py

```diff
@@ -0,0 +1,92 @@
+from src.smart_building_rating_calculator.inputs import (
+    UserInputs,
+    HeatingSource,
+    HotWaterSource,
+)
+from enum import StrEnum
+
+
+class FlexArchetypes(StrEnum):
+    NO_FLEXER = "No Flexer"
+    LOW_TECH_FLEXER = "Low-tech Flexer"
+    GOOD_FLEXER = "Good Flexer"
+    UNTAPPED_FLEXER = "Untapped Flexer"
+    STRONG_FLEXER = "Strong Flexer"
+    GOLD_FLEXER = "Gold Standard Flexer"
+
+
+def calc_flex_archetype(user_inputs: UserInputs, sbr_val: float) -> FlexArchetypes:
```

Comment:
Alternative implementation to Gus's suggestion - abstract each 'if conditions' into a separate function:

- This long if-conditions are very long and hard to read and going to be hard to unit test
- Would be easier to abstract each 'checks' into a function and unit test each function individually
- You also won't need long if-elif-else since you're returning at the end of each condition. You could just:

```
if check_gold_flexer(user_inputs):
     return GOLD_FLEXER.value

if check_strong_flexer(user_inputs):
    return STRONG_FLEXER.value
```

etc. That makes this function a lot smaller, testable, and readable.

Reply:
In reference to @anguschadney's suggestion: Some archetypes depend on `heating_source` being one out of multiple options or `hot_water_source` having a specific value, e.g., to be a "Good Flexer" the household has to have either an EV, battery, HP or electric storage heater. This would mean I would have 4 x options for UserInputs for the user to be a "Good Flexer", i.e.,  good_flexer_1 = UserInputs(...), good_flexer_2 = UserInputs(...), good_flexer_3=, etc. For other flex types, there are even more ways of being that type of flexer. So this makes the suggested logic become quite complex as I would have to write out all the possible options for the user inputs for each archetype. 

To simplify & clarify the code I have instead taken @shengy90's suggestion here. For the functions in `scoring.py`, however, I have taken your suggested approach and created static classes for all the different instances which map to output values.